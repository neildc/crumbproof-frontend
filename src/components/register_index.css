
.registerCard {
    margin: 0 auto;
    text-align: center;
    padding: 10px;
}

.registerForm {
    padding:10px;
}

@media (min-width:768px) {
    .registerCard {
        width: 50%;
        height: 50%;
    }

    .registerForm {
        padding:20px 50px;
    }
}

.registerCardTitle {
    text-align: left;
    backgroundColor: #EEE;
}

.registerButton {
    margin-top: 30px;
    margin-bottom: 10px;
}
