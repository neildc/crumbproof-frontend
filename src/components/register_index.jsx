import "./register_index.css"

import React, { Component } from "react";
import { connect } from "react-redux";
import { authRegister, authClearError } from "../actions/actions_auth.jsx";

import { Field, reduxForm } from "redux-form";

import TextField from 'material-ui/TextField';
import {Card,  CardTitle} from 'material-ui/Card';
import {required} from "../validators.js";

import Snackbar from 'material-ui/Snackbar';
import SubmitButton from "./SubmitButton";

class RegisterIndex extends Component {

  renderField(field) {
      const { meta: { touched, error } } = field;

      return (
          <div>
              <TextField
                  style={{width:"auto"}}
                  hintText=""
                  floatingLabelText={field.label}
                  errorText={touched && error}
                  {...field.input}
                  {...field.custom}
                  type={field.type}
              />
          </div>
      );
  }

  onSubmit(values) {
    return this.props.authRegister(values, () => {
      this.props.history.push("/");
    });
  }

  render() {
    const { handleSubmit, submitting } = this.props;

      return(
          <Card className="registerCard">
            <CardTitle title="Register" className="registerCardTitle"/>

            <form onSubmit={handleSubmit(this.onSubmit.bind(this))}
                  className="registerForm">

                <Field
                    label="Username"
                    name="username"
                    component={this.renderField}
                    validate={[ required ]}
                />
                <Field
                    label="Password"
                    name="password1"
                    type="password"
                    component={this.renderField}
                    validate={[ required ]}
                />

                <Field
                    label="Confirm Password"
                    name="password2"
                    type="password"
                    component={this.renderField}
                    validate={[ required ]}
                />

                <SubmitButton
                  className="registerButton"
                  submittingFlag={submitting}
                  label="Register"
                  labelInProgress="Registering..."
                  fullWidth={true}
                />

            </form>

            <Snackbar
              open={this.props.error != null}
              message={this.props.error ? this.props.error : ""}
              autoHideDuration={5000}
              style={{backgroundColor: "red"}}
              onRequestClose={this.props.authClearError}
            />
          </Card>
    );
  }
}

function mapStateToProps({ auth }, ownProps) {
  return { error: auth.error };
}

export default reduxForm({
  form: "RegisterForm"
})(connect(mapStateToProps, { authRegister, authClearError })(RegisterIndex));
