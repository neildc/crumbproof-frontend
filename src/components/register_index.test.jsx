import React from 'react';
import { shallow } from 'enzyme';
import { RegisterIndex } from './register_index';

it('renders without throwing', () => {
  const component = shallow(
    <RegisterIndex
      error={null}
      handleSubmit={jest.fn()}
      submitting={false}
    />,
  );

  expect(component).toMatchSnapshot();
});
