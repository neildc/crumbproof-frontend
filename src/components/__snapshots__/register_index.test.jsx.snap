// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without throwing 1`] = `
<FadeIn>
  <Card
    className="registerCard"
    expandable={false}
    expanded={null}
    initiallyExpanded={false}
  >
    <CardTitle
      className="registerCardTitle"
      title="Register"
    />
    <form
      className="registerForm"
    >
      <Field
        component={[Function]}
        label="Username"
        name="username"
        validate={
          Array [
            [Function],
          ]
        }
      />
      <Field
        component={[Function]}
        label="Password"
        name="password1"
        type="password"
        validate={
          Array [
            [Function],
          ]
        }
      />
      <Field
        component={[Function]}
        label="Confirm Password"
        name="password2"
        type="password"
        validate={
          Array [
            [Function],
          ]
        }
      />
      <SubmitButton
        className="registerButton"
        fullWidth={true}
        label="Register"
        labelInProgress="Registering..."
        submittingFlag={false}
      />
    </form>
  </Card>
</FadeIn>
`;
